### PR TITLE
Musark 326

### PR DESCRIPTION
**Pull request by jarlah**

### CODE REVIEW

**Comment by jarlah** [comment]
must get it to work with docker

**Comment by jarlah** [comment]
i made it work in docker

File: service_thing_aggregate/src/it/scala/ObjectAggregationIntegrationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      println(response.body)
+      val objects = Json.parse(response.body).validate[Seq[ObjectAggregation]].get
+      val obj = objects.head
+      obj.id mustBe ObjectId(1)
+      obj.displayName mustBe Some("Øks")
+      obj.identifier mustBe MuseumIdentifier("C666", Some("34"))
+    }
+    "get by nodeId that does not exist" in {
+      val nodeId = 99999
+      val response = wsUrl(s"/node/$nodeId/objects").get().futureValue(Timeout(30 seconds))
```
Comment (kpmeen) [inline service_thing_aggregate/src/it/scala/ObjectAggregationIntegrationSpec.scala]: the `Timeout` should be unnecessary when you have the implicit patienceConfig above.

File: service_thing_aggregate/src/it/scala/testHelpers/TestConfigs.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.scalatest.concurrent.PatienceConfiguration
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.time.{Millis, Seconds, Span}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+object TestConfigs {
+
+  trait WaitLonger {
```
Comment (kpmeen) [inline service_thing_aggregate/src/it/scala/testHelpers/TestConfigs.scala]: This didn't work out as expected... remove

File: service_thing_aggregate/src/it/scala/testHelpers/TestConfigs.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "slick.dbs.default.loginTimeout" -> "20000",
+    "slick.dbs.default.socketTimeout" -> "20000",
+    "slick.dbs.default.db.driver" -> "org.h2.Driver",
+    "slick.dbs.default.connectionTestQuery" -> "SELECT 1",
+    "slick.dbs.default.db.url" -> "jdbc:h2:mem:play-test",
+    "slick.dbs.default.leakDetectionThreshold" -> "5000",
+    "evolutionplugin" -> evolve
+  )
+
+  def waitFutureValue[T](fut: Future[T]) = {
```
Comment (kpmeen) [inline service_thing_aggregate/src/it/scala/testHelpers/TestConfigs.scala]: not used...remove

File: service_thing_aggregate/src/it/scala/testHelpers/TestConfigs.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+object TestConfigs {
+
+  trait WaitLonger {
+    implicit val defaultPatience = PatienceConfig(
+      timeout = Span(5, Seconds),
+      interval = Span(500, Millis)
+    )
+  }
+
+  val timeout = PatienceConfiguration.Timeout(15 seconds)
```
Comment (kpmeen) [inline service_thing_aggregate/src/it/scala/testHelpers/TestConfigs.scala]: doesn't belong here....remove

File: service_thing_aggregate/src/main/resources/evolutions/default/1.sql [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ current_location_id  integer,
+ FOREIGN KEY(object_id) REFERENCES MUSIT_MAPPING.VIEW_MUSITTHING(id)
+);
+
+CREATE TABLE MUSARK_STORAGE.STORAGE_NODE(
+ storage_node_id BIGINT NOT NULL  AUTO_INCREMENT,
+ primary key (storage_node_id)
+);
+
+insert INTO MUSARK_STORAGE.STORAGE_NODE(storage_node_id) VALUES(3);
```
Comment (kpmeen) [inline service_thing_aggregate/src/main/resources/evolutions/default/1.sql]: Nitpicking here... but we should be consistent in _how_ we write SQL. Either we use UPPER_CASE for SQL keywords, or we use lower_case for keywords. Agree?

File: service_thing_aggregate/src/main/scala/controllers/ObjectAggregationController.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import services.{ ObjectAggregationService, StorageNodeService }
+
+import scala.concurrent.Future
+
+class ObjectAggregationController @Inject() (
+    service: ObjectAggregationService,
+    storageNodeService: StorageNodeService
+) extends Controller {
+
+  def getObjectsIfNodeExists(nodeId: Long) = Action.async { request =>
```
Comment (kpmeen) [inline service_thing_aggregate/src/main/scala/controllers/ObjectAggregationController.scala]: Somewhat redundantly named, or? Wouldn't get `getObjects` suffice?
If the node with nodeId doesn't exists it directly implies that there are noe objects :-)

File: service_thing_aggregate/src/main/scala/controllers/ObjectAggregationController.scala [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      case MusitSuccess(true) => getObjectsByNodeId(nodeId)
+      case MusitSuccess(false) => Future.successful(NotFound(s"Did not find node with nodeId $nodeId"))
+      case MusitDbError(msg, ex) =>
+        Logger.error(msg, ex.orNull)
+        Future.successful(InternalServerError(msg))
+      case r: MusitError[_] => Future.successful(InternalServerError(r.message))
+    }
+  }
+
+  def getObjectsByNodeId(nodeId: Long): Future[Result] = {
```
Comment (kpmeen) [inline service_thing_aggregate/src/main/scala/controllers/ObjectAggregationController.scala]: make the function `protected` or private to decrease the visibility.

File: service_thing_aggregate/src/main/scala/dao/ObjectAggregationDao.scala [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+) extends HasDatabaseConfigProvider[JdbcProfile] {
+
+  import driver.api._
+
+  def getObjects(nodeId: Long): Future[MusitResult[Seq[ObjectAggregation]]] = {
+    implicit val getObject = GetResult(r =>
+      ObjectAggregation(ObjectId(r.nextLong), MuseumIdentifier.fromSqlString(r.nextString), r.nextStringOption))
+    db.run(
+      sql"""
+         select id, displayId, displayName
```
Comment (kpmeen) [inline service_thing_aggregate/src/main/scala/dao/ObjectAggregationDao.scala:24]: I like this _soooo_ much better for complex queries than the extremely convoluted slick DSL 👍

File: service_thing_aggregate/src/test/scala/testHelpers/TestConfigs.scala [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.time.{ Millis, Seconds, Span }
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+object TestConfigs {
+
+  trait WaitLonger {
+    implicit val defaultPatience = PatienceConfig(
```
Comment (kpmeen) [inline service_thing_aggregate/src/test/scala/testHelpers/TestConfigs.scala]: As with the previous `TestConfigs` file...remove the stuff that isn't used

**Pull request merged**